cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -4,7 +4,64 @@
 # For the licensing terms see $ROOTSYS/LICENSE.
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
-#---Check for Python installation-------------------------------------------------------
+#[[
+The purpose of this machinery is to search Python and set both the ROOT and PyROOT related
+Python variables.
+
+ROOT:
+    - PYTHON_EXECUTABLE (set when Interpreter is found)
+    - PYTHON_VERSION_STRING        ""
+    - PYTHON_VERSION_MAJOR        ""
+    - PYTHON_VERSION_MINOR        ""
+
+PyROOT: a set of specific variables is needed for all the Python versions used to build PyROOT.
+At the time of writing we only have two, called simply Main and Other.
+
+    - PYTHON_EXECUTABLE_Development_Main (set when Interpreter is found)
+    - PYTHON_VERSION_STRING_Development_Main        ""
+    - PYTHON_VERSION_MAJOR_Development_Main        ""
+    - PYTHON_VERSION_MINOR_Development_Main        ""
+    - PYTHON_INCLUDE_DIRS_Development_Main (set when Development is found)
+    - PYTHON_LIBRARIES_Development_Main        ""
+    - PYTHON_UNDER_VERSION_STRING_Development_Main (used to give a version-dependent name to the libraries, to allow
+    multiple builds)
+    - Development_Python${version}_FOUND (set if both Interpreter and Development are found for the specified ${version})
+
+    - PYTHON_EXECUTABLE_Development_Other (set when Interpreter is found)
+    - PYTHON_VERSION_STRING_Development_Other        ""
+    - PYTHON_VERSION_MAJOR_Development_Other        ""
+    - PYTHON_VERSION_MINOR_Development_Other        ""
+    - PYTHON_INCLUDE_DIRS_Development_Other (set when Development is found)
+    - PYTHON_LIBRARIES_Development_Other        ""
+    - PYTHON_UNDER_VERSION_STRING_Development_Other (used to give a version-dependent name to the libraries, to allow
+    multiple builds)
+    - Development_Python${version}_FOUND (set if both Interpreter and Development are found for the specified ${version})
+
+^^^^^^^^
+
+Explanation of the machinery:
+
+The first distinction is based on the CMake version used to build. If it is >= 3.14, than PyROOT can be built
+for multiple Python versions. In case CMake >= 3.14, then we check if PYTHON_EXECUTABLE was specified by the user;
+if so, PyROOT is built with only that version, otherwise an attempt to build it both with Python2 and 3 is done.
+
+If PYTHON_EXECUTABLE is specified:
+    - we check which version it is (from here call X) and call the relative
+    find_package(PythonX COMPONENTS Interpreter Development Numpy)
+    - if Interpreter is found, we set the ROOT related Python variables
+    - if Development is found, we set the PyROOT_Main variables (+ Numpy ones)
```

Comment:
Now they are not `PyROOT_Main`, they are `Development_Main` and they are used both for PyROOT and PyMVA. In another PR (to not trigger new builds) we can correct this.